api/buildings/views.py

```diff
@@ -59,32 +62,40 @@ def public_create(self, request):
                 serializer.errors, status=status.HTTP_400_BAD_REQUEST
             )
 
-
-@api_view(["GET"])
-@permission_classes((permissions.AllowAny,))
-@extend_schema(
-    parameters=[
-        BuildingSearchSerializer,
-        OpenApiParameter(
-            "query", BuildingSearchSerializer, OpenApiParameter.QUERY
-        ),
-    ],
-    description="Some cool building search",
-)
-def building_search(request):
-    query = request.GET.get("query")
-
-    buildings = (
-        Building.approved.annotate(
-            similarity=TrigramSimilarity("address", query)
-        )
-        .filter(similarity__gt=settings.TRIGRAM_SIMILARITY_THRESHOLD)
-        .order_by("-similarity")
+    @extend_schema(
+        parameters=[
+            OpenApiParameter("query", SearchQuerySerializer, OpenApiParameter.QUERY)
+        ],
+        responses=BuildingSearchSerializer
+    )
+    @action(
+        detail=False,
+        permission_classes=[permissions.AllowAny],
     )
+    def search(self, request):
+        """
+        Some cool building search
```

Comment:
This could be changed to something more relevant like `"Search a building by its address"` now that it works